include/DeterministicNoiseSimulator.hpp

```diff
@@ -11,30 +11,58 @@
-        setAmplitudeDampingProbability(noise_prob);
+        initializeNoiseProbabilities(noise_prob);
     }
 
-    void setAmplitudeDampingProbability(double cGateNoiseProbability) {
-        //The probability of amplitude damping (t1) often is double the probability , of phase flip, which is why I double it here
-        noiseProbability                             = cGateNoiseProbability;
-        sqrt_amplitude_damping_probability           = {sqrt(noiseProbability * 2), 0};
-        one_minus_sqrt_amplitude_damping_probability = {sqrt(1 - noiseProbability * 2), 0};
+    DeterministicNoiseSimulator(std::unique_ptr<qc::QuantumComputation>& qc, const std::string& noise_effects, double noise_prob, double amp_noise_prob, double multiQubitGateFactor):
+        qc(qc) {
+        setNoiseEffects(noise_effects);
+        initializeNoiseProbabilities(noise_prob, amp_noise_prob, multiQubitGateFactor);
+    }
+
+    double noiseProb                 = 0.0;
+    double ampDampingProb            = 0.0;
+    double noiseProbSingleQubit      = 0.0;
+    double ampDampingProbSingleQubit = 0.0;
+    double noiseProbMultiQubit       = 0.0;
+    double ampDampingProbMultiQubit  = 0.0;
+
+    void initializeNoiseProbabilities(double cGateNoiseProbability, double cAmplitudeDampingProb = -1, double cMultiQubitGateFactor = 2) {
+        noiseProb            = cGateNoiseProbability;
+        noiseProbSingleQubit = cGateNoiseProbability;
+        noiseProbMultiQubit  = cGateNoiseProbability * cMultiQubitGateFactor;
+
+        if (cAmplitudeDampingProb < 0) {
+            // Default value for amplitude damping prob is double the general error probability
+            ampDampingProb            = cGateNoiseProbability * 2;
+            ampDampingProbSingleQubit = cGateNoiseProbability * 2;
+            ampDampingProbMultiQubit  = cGateNoiseProbability * 2 * cMultiQubitGateFactor;
+        } else {
+            ampDampingProb            = cAmplitudeDampingProb;
+            ampDampingProbSingleQubit = cAmplitudeDampingProb;
+            ampDampingProbMultiQubit  = cAmplitudeDampingProb * cMultiQubitGateFactor;
+        }
+
+        if (noiseProb < 0 || ampDampingProb < 0 || ampDampingProbMultiQubit > 1 || noiseProbMultiQubit > 1) {
+            throw std::runtime_error("Error probabilities are faulty!"
+                                     "\n single qubit error probability: " +
+                                     std::to_string(noiseProbSingleQubit) +
+                                     " multi qubit error probability: " + std::to_string(noiseProbMultiQubit) +
+                                     "\n single qubit amplitude damping  probability: " + std::to_string(ampDampingProbSingleQubit) +
+                                     " multi qubit amplitude damping  probability: " + std::to_string(ampDampingProbMultiQubit));
+        }
     }
 
     std::map<std::string, std::size_t> Simulate([[maybe_unused]] unsigned int shots) override {
```

Comment:
This function should call `DeterministicSimulate` and return a map from basis state to *probability \* shots* for each entry.